### PR TITLE
feat: Add changelog enforcement

### PR DESCRIPTION
**Pull request by Mistermycelium**

### CODE REVIEW

File: .github/workflows/changelog-enforce.yaml [hunk 1]
```diff
@@ -0,0 +1,12 @@
+name: "Changelog Enforcer"
+on:
+  pull_request:
+    types: [opened, synchronize, reopened, ready_for_review, labeled, unlabeled]
```
Comment (fclante) [inline .github/workflows/changelog-enforce.yaml:4]: remove --> ready_for_review, labeled, unlabeled from this array

File: .github/workflows/generate-readme.yaml [hunk 2]
```diff
@@ -4,8 +4,13 @@ name: Generate README
 on:
   workflow_dispatch:
 
+permissions:
+  contents: write
+
 jobs:
   main:
     name: ""
     uses: beeheim/actions-workflows/.github/workflows/generate-readme.yaml@generate-readme
```
Comment (fclante) [inline .github/workflows/generate-readme.yaml:13]: update to reference main, remove the _with:_ block

**Review by fclante — CHANGES_REQUESTED** [review]

**Review by fclante — CHANGES_REQUESTED** [review]
pls fix

File: .github/workflows/changelog-enforce.yaml (hunk 1 above)
Comment (Mistermycelium) [reply .github/workflows/changelog-enforce.yaml]: done, ty

**Review by fclante — APPROVED** [review]

**Pull request merged**